src/hotspot/share/opto/escape.cpp

```diff
@@ -387,6 +398,330 @@ bool ConnectionGraph::compute_escape() {
   return has_non_escaping_obj;
 }
 
+int ConnectionGraph::reduce_allocation_merges() {
+  Unique_Node_List ideal_nodes;
+  ideal_nodes.map(_compile->live_nodes(), NULL);
```

Comment:
You do a full pass over the whole graph to enumerate `Phi`s. Don't you achieve the same when enumerating `LocalVar`s from the CG?